### DISCUSSION TITLE
Operator: `onSubscribe`, `subscriptionTap` to trigger callback at time of subscription

### DISCUSSION THREAD

**Discussion by aYorky** [ReactiveX/rxjs#7436] [Ideas / Feature request]
I recently ran into a scenario where it would have been extremely helpful to know/trigger some side effect when a subscription happens. I looked at `startWith`, but that emits a value, which is not what I was looking for. I needed something like `tap` that doesn't wait for values to be emitted. I realized that Observables fundamentally do not have a way to indicate when they've been subscribed (right? I wouldn't say I'm an expert. 🙂), and there was only one question on SO that I could find with a similar scenario, and I wasn't impressed with the answer. Eventually I created the following custom operator, and I think it could be useful to others as well. I'm not beholden to the name.

```typescript
onSubscribe<T>(f: Function) {
  return (observable: Observable<T>) => {
    return merge(
      of(null).pipe(map(() => f())),
      observable
    ).pipe(skip(1));
  };
}
```

**Comment by demensky** [comment]
Perhaps `tap` [already](https://rxjs.dev/api/index/interface/TapObserver) has the necessary functionality:

```ts
tap({
  subscribe: () => console.log('subscribe'),
  unsubscribe: () => console.log('unsubscribe'),
  finalize: () => console.log('finalize')
})
```

**Reply by aYorky** [reply]
You, sir, have just blown my mind! I am not unfamiliar with the `tap` docs or the operator's usage, yet somehow I have never come across this (tbf, I've never needed this functionality until now). Well, if nothing else, this was an excellent exercise on how to combine operators together to accomplish something, so not a total loss.
